### ISSUE TITLE
Which dependencies are runtime dependencies and which ones are build dependencies?

### ISSUE THREAD

**Issue by vellozzi** [dotnet/coreclr#18097]
I have a question about the dependencies listed in [the coreclr instructions](https://github.com/dotnet/coreclr/blob/release/2.1/Documentation/building/linux-instructions.md) and [the corefx instructions](https://github.com/dotnet/corefx/blob/master/Documentation/building/unix-instructions.md).
Which dependencies are for runtime and which ones are for build-time only?

Also, if I install the SDK and run ldd on all the .so files included, and all the dependencies resolve correctly, does that mean I have everything I need for runtime installed on the box?
In other words, are there any DllImports in the C# for shared libraries other than the ones included in the SDK?

**Comment by RussKeldorph** [comment]
@janvorli

**Comment by janvorli** [comment]
The dependencies listed in the instructions you've mentioned are for building. They list dev packages that contain header files and build time libraries and also some tools that are needed during the build. The runtime dependencies are smaller than that.

Ldd would verify most of the runtime dependencies, but there are two dependencies that are not verified by the ldd. The openssl 1.0 and ICU libraries. We load these using dlopen to handle differences between distros so that our runtime can be portable. For ICU, any version >= 50 works. For openssl, it is important to have openssl 1.0.x installed, openssl 1.1 is incompatible (although the version would suggest it is a minor change, there is in fact a huge difference between those).

You can see the runtime dependencies listed here:
https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x
But for 2.1, some of those dependencies are not needed anymore or are just optional.
libunwind and libuuid dependency is gone and the libcurl is now optional. The liblttng is needed only if you want to use tracing.

**Comment by vellozzi** [comment]
Thanks.  That helps a lot.

**Comment by janvorli** [comment]
@vellozzi the PAL tests actually don't cover any of those.
